landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+            else:
+                self._A[i, i] = -(self._alpha - (self._alpha * alphas_fractios[0]))
+                cnti = i - 1  # rows,
+                cnt = 1
+                while cnti >= 0 and cnt <= (len(alphas_fractios) - 1):
+                    self._A[cnti, i] = self._alpha * alphas_fractios[cnt]
+                    if cnti == 0 and cnt <= (len(alphas_fractios) - 1):
+                        self._A[cnti, i] = (1 - alphas_fractios[0]) - np.sum(
+                            alphas_fractios[1:cnt]
+                        )
+                        cnt += 1
+                        cnti -= 1
+                    cnt += 1
+                    cnti -= 1
+
+    def set_grading_classes(self):
+
+        input_sizes_flag = self._input_sizes_flag
+
+        def lower_limit_of(maxsize):
+            lower_limit = maxsize * (1 / self._N) ** (power_of)
+            return lower_limit
+
+        upperlimits = []
+        lowerlimits = []
+        num_of_size_classes_plusone = self._n_sizes + 1
+
+        if input_sizes_flag:
+            meansizes = self._meansizes
+            self._meansizes = np.array(meansizes)
+            self._upperlims = np.array(meansizes)
+            self._lowerlims = np.insert(np.array(meansizes), 0, 0)
+        else:
+            maxsize = self._grain_max_size
+            power_of = self._power_of
+            for _ in range(num_of_size_classes_plusone - 1):
+                upperlimits.append(maxsize)
+                maxsize = lower_limit_of(maxsize)
+                lowerlimits.append(maxsize)
+
+            self._upperlims = np.sort(upperlimits)
+            self._lowerlims = np.sort(lowerlimits)
+            self._meansizes = (
+                np.array(self._upperlims) + np.array(self._lowerlims)
+            ) / 2
+
+        self._grid.at_node["grains_classes__size"][self._grid.nodes] *= self._meansizes
+
+    def create_dist(
+        self, median_size=None, std=None, is_bedrock_distribution_flag=False
+    ):
+        """
+        This procedure split the total soil weight between all size classes
+        assuming normal distribution around the median size class. Different
+        grain size distribution can be assigned to the initial soil layer and
+        the bedrock layer
+        """
+
+        if not is_bedrock_distribution_flag:
+
+            is_bedrock_distribution_flag = self._is_bedrock_distribution_flag
+            median_size = self._initial_median_size
+            std = self._std
+            total_soil_weight = self._initial_total_soil_weight
+            grains_weight__distribution = self._generate_normal_distribution(
+                median_size=median_size, std=std, total_soil_weight=total_soil_weight
+            )
+
+        else:
+            total_bedrock_weight = 10000  # There is no meaning for this number
+            if median_size is None:
+                grains_weight__distribution = self.g_state0
+            else:
+                median_size = median_size
```

Comment:
Remove.

Reply:
I removed `median_size = media_size`.
I think the `else` block is necessary here to differentiate between cases where `median_size` is provided or not. Without the `else` block, `grains_weight__distribution` will be rewritten. Am I missing something?

Reply:
@yuvalsmilo You're correct. I guess I wan't wearing my glasses and got the indentation mixed up!